File: scripts/screenshot.js
const puppeteer = require('puppeteer');

const url = process.env.SCREENSHOT_URL || process.argv[2] || 'http://localhost:4000';
const output = process.env.SCREENSHOT_OUTPUT || process.argv[3] || 'screenshot.png';
const width = parseInt(process.env.SCREENSHOT_WIDTH || '1280', 10);
const height = parseInt(process.env.SCREENSHOT_HEIGHT || '800', 10);

(async () => {
  const browser = await puppeteer.launch({ args: ['--no-sandbox', '--disable-setuid-sandbox'] });
  const page = await browser.newPage();
  await page.setViewport({ width, height });
  await page.goto(url, { waitUntil: 'networkidle2' });
  await page.screenshot({ path: output, fullPage: false });
  await browser.close();
  console.log('Saved screenshot to', output);
})();

File: scripts/screenshot.py
import os
from playwright.sync_api import sync_playwright

url = os.environ.get("SCREENSHOT_URL", "http://localhost:4000")
output = os.environ.get("SCREENSHOT_OUTPUT", "screenshot.png")
width = int(os.environ.get("SCREENSHOT_WIDTH", "1280"))
height = int(os.environ.get("SCREENSHOT_HEIGHT", "600"))

with sync_playwright() as p:
    browser = p.chromium.launch()
    context = browser.new_context(viewport={"width": width, "height": height})
    page = context.new_page()
    page.goto(url, wait_until="networkidle")
    page.screenshot(path=output, full_page=False)
    browser.close()
    print(f"Saved screenshot to {output}")
